llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -1377,7 +1377,7 @@ bool MachineLICMImpl::IsProfitableToHoist(MachineInstr &MI,
                }) &&
         IsLoopInvariantInst(MI, CurLoop) &&
         any_of(MRI->use_nodbg_instructions(DefReg),
-               [&CurLoop, this, DefReg, Cost](MachineInstr &UseMI) {
+               [&CurLoop, this, DefReg, std::move(Cost)](MachineInstr &UseMI) {
```

Comment:
If you want to do this you need to make use of move captures https://stackoverflow.com/a/20669290

This is only really supported in C++14 and higher. It also seems like this type of change could probably be made throughout the whole codebase. Is there any real reason to just focus on this one line?

Reply:
Hey, I've made multiple PRs (grouped by component) because it gets hard to get them reviewed in different components and the bot pings a lot of people. If you can review them (and possibly merge?) I can close all the others and make a new one.

Reply:
That's ok. I think this change is small enough to not warrant that.

Reply:
This change still needs to make use of lambda captures though. I'm happy to approve once that's done